core/app_test.go

```diff
@@ -262,6 +262,23 @@ func TestPidEnvVar(t *testing.T) {
 	}
 }
 
+func TestControlServerConfigParse(t *testing.T) {
+	jsonFragment := `{
+    "consul": "consul:8500",
+    "control": {
+	"socket": "/var/run/cp.socket"
+    }
```

Comment:
As of right now, this is required for the socket to exist. I need to add the default case so the socket always comes up.